mpas_analysis/ocean/nino34_index.py

```diff
@@ -141,15 +141,19 @@ def compute_nino34_index(regionSST, calendar):  # {{{
     03/23/2017
     """
 
-    if not isinstance(regionSST, xr.core.dataarray.DataArray):
-        raise ValueError('regionSST should be an xarray DataArray')
+    if not isinstance(ds, xr.core.dataset.Dataset):
+        raise ValueError('ds should be an xarray Dataset')
+
+    ds = climatology.add_months_and_days_in_month(ds, calendar)
```

Comment:
So the problem here is that we explicitly do a `groupby('month')` below.  It is not obvious where `month` got added to the data set if this is not made explicit with an exposed call.  In fact, I changed the behavior so `month` *doesn't* get added to `ds` behind the scenes in `compute*_climatology` because I realized this behavior would be unexpected.  As a result, any code that needs `month` to be added to `ds` for use outside of `climatology` really needs to do it manually themselves.